Update dependencies Jan. 2022

## Situation

When used in combination with the latest BucketFS Java 2.2.0, there is a version conflict in:
* `org.glassfish:jakarta.json`
* `jakarta.json.bind:jakarta.json.bind`

### Symptoms

Error message:
```
org.opentest4j.AssertionFailedError: Unexpected exception thrown: javax.json.JsonException: Provider org.glassfish.json.JsonProviderImpl could not be instantiated: java.lang.ClassCastException: class org.glassfish.json.JsonProviderImpl cannot be cast to class javax.json.spi.JsonProvider (org.glassfish.json.JsonProviderImpl and javax.json.spi.JsonProvider are in unnamed module of loader 'app')
	at com.exasol.containertest.featuretests.IT_SyncBugs_2.<init>(IT_SyncBugs_2.java:23)
Caused by: javax.json.JsonException: Provider org.glassfish.json.JsonProviderImpl could not be instantiated: java.lang.ClassCastException: class org.glassfish.json.JsonProviderImpl cannot be cast to class javax.json.spi.JsonProvider (org.glassfish.json.JsonProviderImpl and javax.json.spi.JsonProvider are in unnamed module of loader 'app')
	at com.exasol.containertest.featuretests.IT_SyncBugs_2.<init>(IT_SyncBugs_2.java:23)
Caused by: java.lang.ClassCastException: class org.glassfish.json.JsonProviderImpl cannot be cast to class javax.json.spi.JsonProvider (org.glassfish.json.JsonProviderImpl and javax.json.spi.JsonProvider are in unnamed module of loader 'app')
	at com.exasol.containertest.featuretests.IT_SyncBugs_2.<init>(IT_SyncBugs_2.java:23)
```

## Acceptance Criteria

* Dependencies updated to the latest stable available on 2022-01-11

--- a/dependencies.md
+++ b/dependencies.md
@@ -3,74 +3,80 @@
 
 ## Compile Dependencies
 
-| Dependency                                      | License                                                  |
-| ----------------------------------------------- | -------------------------------------------------------- |
-| [JSR 374 (JSON Processing) API][0]              | [Dual license consisting of the CDDL v1.1 and GPL v2][1] |
-| [JSR 374 (JSON Processing) Default Provider][0] | [Dual license consisting of the CDDL v1.1 and GPL v2][1] |
-| [error-reporting-java][4]                       | [MIT][5]                                                 |
+| Dependency                       | License                                                                                                      |
+| -------------------------------- | ------------------------------------------------------------------------------------------------------------ |
+| [Jakarta JSON Processing API][0] | [Eclipse Public License 2.0][1]; [GNU General Public License, version 2 with the GNU Classpath Exception][2] |
+| [error-reporting-java][3]        | [MIT][4]                                                                                                     |
 
 ## Test Dependencies
 
 | Dependency                | License                          |
 | ------------------------- | -------------------------------- |
-| [Hamcrest][6]             | [BSD License 3][7]               |
-| [JUnit Jupiter Engine][8] | [Eclipse Public License v2.0][9] |
-| [JUnit Jupiter Params][8] | [Eclipse Public License v2.0][9] |
+| [Hamcrest][5]             | [BSD License 3][6]               |
+| [JUnit Jupiter Engine][7] | [Eclipse Public License v2.0][8] |
+| [JUnit Jupiter Params][7] | [Eclipse Public License v2.0][8] |
+
+## Runtime Dependencies
+
+| Dependency                   | License                                                                                                      |
+| ---------------------------- | ------------------------------------------------------------------------------------------------------------ |
+| [JSON-P Default Provider][0] | [Eclipse Public License 2.0][1]; [GNU General Public License, version 2 with the GNU Classpath Exception][2] |
 
 ## Plugin Dependencies
 
 | Dependency                                              | License                                        |
 | ------------------------------------------------------- | ---------------------------------------------- |
-| [Maven Surefire Plugin][12]                             | [Apache License, Version 2.0][13]              |
-| [JaCoCo :: Maven Plugin][14]                            | [Eclipse Public License 2.0][15]               |
-| [Apache Maven Source Plugin][16]                        | [Apache License, Version 2.0][13]              |
-| [Apache Maven Compiler Plugin][18]                      | [Apache License, Version 2.0][13]              |
-| [Versions Maven Plugin][20]                             | [Apache License, Version 2.0][13]              |
-| [org.sonatype.ossindex.maven:ossindex-maven-plugin][22] | [ASL2][23]                                     |
-| [Apache Maven Enforcer Plugin][24]                      | [Apache License, Version 2.0][13]              |
-| [Apache Maven Javadoc Plugin][26]                       | [Apache License, Version 2.0][13]              |
-| [Apache Maven GPG Plugin][28]                           | [Apache License, Version 2.0][23]              |
-| [Project keeper maven plugin][30]                       | [MIT][5]                                       |
-| [error-code-crawler-maven-plugin][32]                   | [MIT][5]                                       |
-| [Reproducible Build Maven Plugin][34]                   | [Apache 2.0][23]                               |
-| [Maven Failsafe Plugin][36]                             | [Apache License, Version 2.0][13]              |
-| [Maven Deploy Plugin][38]                               | [The Apache Software License, Version 2.0][23] |
-| [Nexus Staging Maven Plugin][40]                        | [Eclipse Public License][41]                   |
-| [Maven Clean Plugin][42]                                | [The Apache Software License, Version 2.0][23] |
-| [Maven Resources Plugin][44]                            | [The Apache Software License, Version 2.0][23] |
-| [Maven JAR Plugin][46]                                  | [The Apache Software License, Version 2.0][23] |
-| [Maven Install Plugin][48]                              | [The Apache Software License, Version 2.0][23] |
-| [Maven Site Plugin 3][50]                               | [The Apache Software License, Version 2.0][23] |
+| [Maven Surefire Plugin][14]                             | [Apache License, Version 2.0][15]              |
+| [JaCoCo :: Maven Plugin][16]                            | [Eclipse Public License 2.0][17]               |
+| [Apache Maven Source Plugin][18]                        | [Apache License, Version 2.0][15]              |
+| [Apache Maven Compiler Plugin][20]                      | [Apache License, Version 2.0][15]              |
+| [Versions Maven Plugin][22]                             | [Apache License, Version 2.0][15]              |
+| [org.sonatype.ossindex.maven:ossindex-maven-plugin][24] | [ASL2][25]                                     |
+| [Apache Maven Enforcer Plugin][26]                      | [Apache License, Version 2.0][15]              |
+| [Apache Maven Javadoc Plugin][28]                       | [Apache License, Version 2.0][15]              |
+| [Apache Maven GPG Plugin][30]                           | [Apache License, Version 2.0][25]              |
+| [Project keeper maven plugin][32]                       | [MIT][4]                                       |
+| [error-code-crawler-maven-plugin][34]                   | [MIT][4]                                       |
+| [Reproducible Build Maven Plugin][36]                   | [Apache 2.0][25]                               |
+| [Maven Failsafe Plugin][38]                             | [Apache License, Version 2.0][15]              |
+| [Maven Deploy Plugin][40]                               | [The Apache Software License, Version 2.0][25] |
+| [Nexus Staging Maven Plugin][42]                        | [Eclipse Public License][43]                   |
+| [Maven Clean Plugin][44]                                | [The Apache Software License, Version 2.0][25] |
+| [Maven Resources Plugin][46]                            | [The Apache Software License, Version 2.0][25] |
+| [Maven JAR Plugin][48]                                  | [The Apache Software License, Version 2.0][25] |
+| [Maven Install Plugin][50]                              | [The Apache Software License, Version 2.0][25] |
+| [Maven Site Plugin 3][52]                               | [The Apache Software License, Version 2.0][25] |
 
-[30]: https://github.com/exasol/project-keeper-maven-plugin
-[4]: https://github.com/exasol/error-reporting-java
-[0]: https://javaee.github.io/jsonp
-[23]: http://www.apache.org/licenses/LICENSE-2.0.txt
-[12]: https://maven.apache.org/surefire/maven-surefire-plugin/
-[40]: http://www.sonatype.com/public-parent/nexus-maven-plugins/nexus-staging/nexus-staging-maven-plugin/
-[42]: http://maven.apache.org/plugins/maven-clean-plugin/
-[5]: https://opensource.org/licenses/MIT
-[36]: https://maven.apache.org/surefire/maven-failsafe-plugin/
-[20]: http://www.mojohaus.org/versions-maven-plugin/
-[7]: http://opensource.org/licenses/BSD-3-Clause
-[18]: https://maven.apache.org/plugins/maven-compiler-plugin/
-[1]: https://oss.oracle.com/licenses/CDDL+GPL-1.1
-[28]: http://maven.apache.org/plugins/maven-gpg-plugin/
-[15]: https://www.eclipse.org/legal/epl-2.0/
-[41]: http://www.eclipse.org/legal/epl-v10.html
-[14]: https://www.jacoco.org/jacoco/trunk/doc/maven.html
-[34]: http://zlika.github.io/reproducible-build-maven-plugin
-[46]: http://maven.apache.org/plugins/maven-jar-plugin/
-[13]: https://www.apache.org/licenses/LICENSE-2.0.txt
-[24]: https://maven.apache.org/enforcer/maven-enforcer-plugin/
-[9]: https://www.eclipse.org/legal/epl-v20.html
-[48]: http://maven.apache.org/plugins/maven-install-plugin/
-[8]: https://junit.org/junit5/
-[22]: https://sonatype.github.io/ossindex-maven/maven-plugin/
-[16]: https://maven.apache.org/plugins/maven-source-plugin/
-[6]: http://hamcrest.org/JavaHamcrest/
-[38]: http://maven.apache.org/plugins/maven-deploy-plugin/
-[50]: http://maven.apache.org/plugins/maven-site-plugin/
-[44]: http://maven.apache.org/plugins/maven-resources-plugin/
-[26]: https://maven.apache.org/plugins/maven-javadoc-plugin/
-[32]: https://github.com/exasol/error-code-crawler-maven-plugin
+[32]: https://github.com/exasol/project-keeper-maven-plugin
+[3]: https://github.com/exasol/error-reporting-java
+[25]: http://www.apache.org/licenses/LICENSE-2.0.txt
+[14]: https://maven.apache.org/surefire/maven-surefire-plugin/
+[42]: http://www.sonatype.com/public-parent/nexus-maven-plugins/nexus-staging/nexus-staging-maven-plugin/
+[44]: http://maven.apache.org/plugins/maven-clean-plugin/
+[4]: https://opensource.org/licenses/MIT
+[38]: https://maven.apache.org/surefire/maven-failsafe-plugin/
+[22]: http://www.mojohaus.org/versions-maven-plugin/
+[6]: http://opensource.org/licenses/BSD-3-Clause
+[20]: https://maven.apache.org/plugins/maven-compiler-plugin/
+[30]: http://maven.apache.org/plugins/maven-gpg-plugin/
+[17]: https://www.eclipse.org/legal/epl-2.0/
+[43]: http://www.eclipse.org/legal/epl-v10.html
+[16]: https://www.jacoco.org/jacoco/trunk/doc/maven.html
+[36]: http://zlika.github.io/reproducible-build-maven-plugin
+[48]: http://maven.apache.org/plugins/maven-jar-plugin/
+[1]: https://projects.eclipse.org/license/epl-2.0
+[15]: https://www.apache.org/licenses/LICENSE-2.0.txt
+[26]: https://maven.apache.org/enforcer/maven-enforcer-plugin/
+[8]: https://www.eclipse.org/legal/epl-v20.html
+[50]: http://maven.apache.org/plugins/maven-install-plugin/
+[7]: https://junit.org/junit5/
+[24]: https://sonatype.github.io/ossindex-maven/maven-plugin/
+[0]: https://github.com/eclipse-ee4j/jsonp
+[18]: https://maven.apache.org/plugins/maven-source-plugin/
+[2]: https://projects.eclipse.org/license/secondary-gpl-2.0-cp
+[5]: http://hamcrest.org/JavaHamcrest/
+[40]: http://maven.apache.org/plugins/maven-deploy-plugin/
+[52]: http://maven.apache.org/plugins/maven-site-plugin/
+[46]: http://maven.apache.org/plugins/maven-resources-plugin/
+[28]: https://maven.apache.org/plugins/maven-javadoc-plugin/
+[34]: https://github.com/exasol/error-code-crawler-maven-plugin

--- a/doc/changes/changelog.md
+++ b/doc/changes/changelog.md
@@ -1,5 +1,6 @@
 # Changes
 
+* [0.2.2](changes_0.2.2.md)
 * [0.2.1](changes_0.2.1.md)
 * [0.2.0](changes_0.2.0.md)
 * [0.1.0](changes_0.1.0.md)

--- a/doc/changes/changes_0.2.2.md
+++ b/doc/changes/changes_0.2.2.md
@@ -1,0 +1,19 @@
+# exasol-version-number-provider 0.2.2, released 2021-01-11
+
+Code name: Dependency Updates
+
+## Summary
+
+In this release we updated the json dependencies.
+
+## Dependency Updates
+
+### Compile Dependency Updates
+
+* Added `jakarta.json:jakarta.json-api:2.0.1`
+* Removed `javax.json:javax.json-api:1.1.4`
+* Removed `org.glassfish:javax.json:1.1.4`
+
+### Runtime Dependency Updates
+
+* Added `org.glassfish:jakarta.json:2.0.1`

--- a/pom.xml
+++ b/pom.xml
@@ -4,7 +4,7 @@
     <modelVersion>4.0.0</modelVersion>
     <groupId>com.exasol</groupId>
     <artifactId>exasol-version-number-provider</artifactId>
-    <version>0.2.1</version>
+    <version>0.2.2</version>
     <name>exasol-version-number-provider</name>
     <description>Tool that always provides you the most recent version of the Exasol docker-db.</description>
     <url>https://github.com/exasol/exasol-version-number-provider</url>

--- a/pom.xml
+++ b/pom.xml
@@ -74,14 +74,15 @@
     </repositories>
     <dependencies>
         <dependency>
-            <groupId>javax.json</groupId>
-            <artifactId>javax.json-api</artifactId>
-            <version>1.1.4</version>
+            <groupId>jakarta.json</groupId>
+            <artifactId>jakarta.json-api</artifactId>
+            <version>2.0.1</version>
         </dependency>
         <dependency>
             <groupId>org.glassfish</groupId>
-            <artifactId>javax.json</artifactId>
-            <version>1.1.4</version>
+            <artifactId>jakarta.json</artifactId>
+            <version>2.0.1</version>
+            <scope>runtime</scope>
         </dependency>
         <dependency>
             <groupId>com.exasol</groupId>

--- a/src/main/java/com/exasol/versionnumberprovider/ExasolVersionNumberProvider.java
+++ b/src/main/java/com/exasol/versionnumberprovider/ExasolVersionNumberProvider.java
@@ -4,7 +4,7 @@ import java.util.Comparator;
 import java.util.List;
 import java.util.stream.Collectors;
 
-import javax.json.JsonArray;
+import jakarta.json.JsonArray;
 
 /**
  * This class gives you access to the available Exasol docker-db version numbers.

--- a/src/main/java/com/exasol/versionnumberprovider/ExasolVersionNumberProviderFactory.java
+++ b/src/main/java/com/exasol/versionnumberprovider/ExasolVersionNumberProviderFactory.java
@@ -5,8 +5,8 @@ import java.io.InputStream;
 import java.net.MalformedURLException;
 import java.net.URL;
 
-import javax.json.Json;
-import javax.json.JsonReader;
+import jakarta.json.Json;
+import jakarta.json.JsonReader;
 
 import com.exasol.errorreporting.ExaError;
 

--- a/src/test/java/com/exasol/versionnumberprovider/LocalExasolVersionNumberProviderFactory.java
+++ b/src/test/java/com/exasol/versionnumberprovider/LocalExasolVersionNumberProviderFactory.java
@@ -3,8 +3,8 @@ package com.exasol.versionnumberprovider;
 import java.io.IOException;
 import java.io.InputStream;
 
-import javax.json.Json;
-import javax.json.JsonReader;
+import jakarta.json.Json;
+import jakarta.json.JsonReader;
 
 public class LocalExasolVersionNumberProviderFactory {
 

